dali/python/backend_impl.cc

```diff
@@ -111,8 +112,82 @@ py::dict ArrayInterfaceRepr(Tensor<Backend> &t) {
   return d;
 }
 
+template<typename SrcBackend>
+TensorListShape<> CreateShape(TensorShape<> &shape, TensorList<SrcBackend>*) {
```

Comment:
I think this name is too generic and the usage of unused parameter's type to determine behavior is quite obscure. Can't you have a dedicated function for each case?
At the very least please do something like:
```suggestion
TensorListShape<> ConvertShape(const TensorShape<> &shape, TensorList<SrcBackend> *shape_type_placeholder) {
```

Reply:
It is used in FillTensorFromDlPack which may handle Tensor and TensorList.